packages/contracts/src/timelock/ITimelockAsyncUnlock.sol

```diff
@@ -6,27 +6,30 @@ pragma solidity ^0.8.20;
  */
 interface ITimelockAsyncUnlock {
     /**
-     * @dev Requests unlock amount for depositPeriod
+     * @dev Requests unlock for multiple deposit periods and amounts
      *
-     * @param owner The address that owns tokens to be unlocked
-     * @param depositPeriod the id of multi token
-     * @param amount the amount of token to be requested for unlock
+     * @param owner The address of the token owner requesting the unlock
+     * @param depositPeriods The IDs of the deposit periods for which tokens are being requested to unlock
+     * @param amounts The amounts of tokens to unlock for each respective deposit period.
      *
-     * @return unlockPeriod the time period when owner can unlock
+     * @return requestId The ID of the unlock request and it can be used in unlock
      */
-    function requestUnlock(address owner, uint256 depositPeriod, uint256 amount)
+    function requestUnlock(address owner, uint256[] memory depositPeriods, uint256[] memory amounts)
         external
-        returns (uint256 unlockPeriod);
+        returns (uint256 requestId);
 
     /**
-     * @dev Unlocks amount for depositPeriod and unlockPeriod
+     * @dev Unlocks amount using requestId
      *
-     * @param owner The address that owns tokens to be unlocked
-     * @param depositPeriod the id of multi token
-     * @param unlockPeriod the time period when owner can unlock
-     * @param amount the amount of token to be unlocked
+     * @param owner The address of the token owner who made the unlock request
+     * @param requestId The ID of the unlock request generated by `requestUnlock`
+     *
+     * @return depositPeriods The deposit periods of tokens that are unlocked
+     * @return amounts The amounts of tokens for each respective deposit period that are unlocked
      */
-    function unlock(address owner, uint256 depositPeriod, uint256 unlockPeriod, uint256 amount) external;
```

Comment:
please keep the "single" version of the requestUnlock and add a requestUnlockBatch.  similar to ERC1155 balanceOf() and balanceOfBatch().  the requestUnlockBatch should just iterate through periods and call to to requestUnlock 

    function requestUnlock(address owner, uint256 depositPeriod, uint256 amount)
        external
        returns (uint256 unlockPeriod);